spec/Section 6 -- Execution.md

```diff
@@ -479,17 +479,28 @@ The depth-first-search order of the field groups produced by {CollectFields()}
 is maintained through execution, ensuring that fields appear in the executed
 response in a stable and predictable order.
 
+When {CollectFields()} is used during validation (see for example the
+[single root field](#sec-Single-root-field) subscription operation validation
+rule), the runtime value for {variableValues} will not be available - in this
+case we set {variableValues} to {null} and forbid the use of the `@skip` and
+`@include` directives. During execution, {variableValues} will always be
+non-null.
+
 CollectFields(objectType, selectionSet, variableValues, visitedFragments):
 
   * If {visitedFragments} is not provided, initialize it to the empty set.
   * Initialize {groupedFields} to an empty ordered map of lists.
   * For each {selection} in {selectionSet}:
-    * If {selection} provides the directive `@skip`, let {skipDirective} be that directive.
-      * If {skipDirective}'s {if} argument is {true} or is a variable in {variableValues} with the value {true}, continue with the next
-      {selection} in {selectionSet}.
-    * If {selection} provides the directive `@include`, let {includeDirective} be that directive.
-      * If {includeDirective}'s {if} argument is not {true} and is not a variable in {variableValues} with the value {true}, continue with the next
-      {selection} in {selectionSet}.
+    * If {variableValues} is {null}:
+      * {selection} must not provide the `@skip` directive.
+      * {selection} must not provide the `@include` directive.
```

Comment:
This is breaking. We already allow `@skip(if: $var)` with no variables provided to result in not skipping.

Reply:
This is only breaking in the way that it was intended to be breaking; I checked every call to CollectFields when I proposed the change and in all cases variableValues was non-null. The new `null` was introduced by me in this one specific location (only intended to be called from validation).